### PR TITLE
Port to Python CLI

### PR DESCRIPTION
**Pull request by mariamrf**
- Flow:
    - `init`: initialize keylist.
    - `generate`: generate password either using existing config or new one if non existent
    - `reset`: reset password config
- Included README and some tests (not super comprehensive tbh..)
- Also included: Github Actions workflow to deploy to PyPI on merge to `main`. I am managing the secrets on my fork and will test it there shortly. Afterwards I can transfer ownership so the secrets can be maintained here.

### CODE REVIEW

**Comment by mariamrf** [comment]
There's an issue with the imports. I'll fix it and mark it as ready when it is.

**Comment by nourselim0** [comment]
Thank you so much for your effort, you really went beyond here 😁 (unit tests? PyPI with installing the script in PATH? wow!)

I'll test this over on the weekend, specifically, I'll take my own key list files and make sure the master check and password generation is identical.

In the mean time, can you squash the commits a bit, it feels like after the first 2 commits, it was mostly fixes (typos and so on), so you can do an interactive rebase and reorder commits and mark them as "fixup", so the end result would just be the first 2 commits. (I'm a bit OCD with commits, I know 😅)

**Comment by mariamrf** [comment]
Commits squashed so it's only 2 now~

File: passgen-py/src/utils.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+def check_master(password, keylist):
+    hashed = hash_master(
+        password, keylist.master.salt, keylist.master.iter_count)
+    return hashed == keylist.master.hashed
+
+
+def hash_master(password, salt, iter_count):
+    # salt is assumed to be base64 encoded
+    salt = base64.b64decode(salt)
+    hashed = pbkdf2_hmac('sha256', password.encode(), salt, iter_count)
```
Comment (nourselim0) [inline passgen-py/src/utils.py:21]:
```py
hashed = b''
for _ in range(iter_count):
    hashed = sha256(hashed + password.encode() + salt).digest()
```
and remove the now unusued pbkdf2_hmac import

**Review by nourselim0 — CHANGES_REQUESTED** [review]
hey, I finally got around to reviewing this PR 😅
So I had to do two changes to make it generate identical password to the dotnet-core version and pass master check for files generated by the dotnet-core version.
First I made the Label case-sensitive (remove the `.lower()` lines), second is I changed the `hash_master` function because the result of PBKDF2 was different from the repeated SHA256 I was doing manually.
I realize now that I should really be using PBKDF2 or some other standard key derivation function (maybe Argon2?) as my iteration might actually be making things less secure... I failed to follow my own advice of "never roll your own crypto function", anyways, I'll probably do that for version 2 of the key file 😅

I'm not sure how I can push that commit I made, this is the first time I accept a PR from a branch on a repo I don't have access to, so I just wrote the `hash_master` code change as PR line comment.

**Review by nourselim0 — APPROVED** [review]
Ok so I updated my master hashing algorithm to PBKDF2 😅
I'll just merge this then do a commit to make the label case-sensitive 😁

**Pull request merged**